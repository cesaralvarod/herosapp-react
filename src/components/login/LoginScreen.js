import React, { useContext } from "react";
import { AuthContext } from "../../auth/AuthContext";
import { types } from "../../types/types";

const LoginScreen = ({ history }) => {
  const { dispatch } = useContext(AuthContext);

  const lastPath = localStorage.getItem("lastPath") || "/";

  const handleClick = () => {
    dispatch({
      type: types.login,
      payload: {
        name: "Cesar",
      },
    });

    history.replace(lastPath);
  };

  return (
    <div className="container">
      <h1>Login</h1>
      <hr />
      <button className="btn btn-primary" onClick={handleClick}>
        Login
      </button>
    </div>
  );
};

export default LoginScreen;
