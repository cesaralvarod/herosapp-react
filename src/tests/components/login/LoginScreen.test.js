import React from "react";
import { mount } from "enzyme";
import LoginScreen from "../../../components/login/LoginScreen";
import { AuthContext } from "../../../auth/AuthContext";

describe("Testing in LoginScreen", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      name: "Cesar",
      logged: true,
    },
  };

  const history = { replace: jest.fn() };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <LoginScreen history={history} />
    </AuthContext.Provider>
  );

  test("Component should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should simulate click in login button", () => {
    const btnLogin = wrapper.find("button.btn");

    btnLogin.simulate("click");

    expect(contextValue.dispatch).toHaveBeenCalledTimes(1);
  });
});
