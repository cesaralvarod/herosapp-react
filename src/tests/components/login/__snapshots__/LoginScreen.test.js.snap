// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in LoginScreen Component should render correctly 1`] = `
<div
  className="container"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
